Mathlib/NumberTheory/ModularForms/QExpansion.lean

```diff
@@ -38,6 +38,16 @@ theorem Function.Periodic.im_invQParam_pos_of_abs_lt_one
     (div_neg_of_neg_of_pos (neg_lt_zero.mpr hh) Real.two_pi_pos)
     ((Real.log_neg_iff (Complex.abs.pos hq_ne)).mpr hq)
 
+open Real in
+lemma qParam_im_ge_half (ξ : ℍ) (hξ : 1 / 2 ≤ ξ.im) : ‖𝕢 1 ξ‖ ≤ rexp (-(π * √3 * (1 / 2))) := by
```

Comment:
Why the constant  `rexp (-(π * √3 * (1 / 2)))`? If you assume `1 / 2 ≤ ξ.im` then the best possible bound is clearly `rexp (-π)` which is both a stronger and a simpler statement.

(The reason `√3` seems relevant is because the lemma `exists_one_half_le_im_and_norm_le` is not optimal; it gives us `ξ` with `1 / 2 ≤ ξ.im`, but actually one can find `ξ` with `√3 / 2 ≤ ξ.im`, and then we would have `‖𝕢 1 ξ‖ ≤ rexp (-(π * √3))`. I deliberately omitted the `√3` factor, because the weaker statement is sufficient and makes the arithmetic easier. You seem to have weakened the statement further while at the same time making it more complicated.

Reply:
(You could for instance show
```
lemma Function.Periodic.qParam_abs_le_of_one_half_le_abs {ξ : ℂ} (hξ : 1 / 2 ≤ ξ.im) :
    ‖𝕢 1 ξ‖ ≤ rexp (-π) := by
  rwa [Periodic.qParam, ofReal_one, div_one, Complex.norm_eq_abs, Complex.abs_exp, Real.exp_le_exp,
    mul_right_comm, mul_I_re, neg_le_neg_iff, ← ofReal_ofNat, ← ofReal_mul, im_ofReal_mul,
    mul_comm _ π, mul_assoc, le_mul_iff_one_le_right Real.pi_pos, ← div_le_iff₀' (by norm_num)]
```

Reply:
Thanks! I was just not thinking enough about it!